lib/network_utils.pm

```diff
@@ -0,0 +1,49 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Functional methods to operate on network
+# Maintainer: Rodion Iafarov <riafarov@suse.com>
+package networkbase;
+
+use strict;
+use Exporter;
+
+our @EXPORT = qw(setup_static_network check_and_recover_network);
+
+use testapi;
+
+=head2 setup_static_network
+Configure static IP on SUT with setting up default GW.
+Also doing test ping to 10.0.2.2 to check that network is alive
+=cut
+sub setup_static_network {
+    my ($ip, $gw) = @_;
+    $gw ||= '10.0.2.2';
+    assert_script_run("echo 'default $gw - -' > /etc/sysconfig/network/routes");
+    assert_script_run("echo 'nameserver 10.160.0.1' >> /etc/resolv.conf");
+    my $iface = script_output('ls /sys/class/net/ | grep -v lo | head -1');
+    assert_script_run qq(echo -e "\\nSTARTMODE='auto'\\nBOOTPROTO='static'\\nIPADDR='$ip'">/etc/sysconfig/network/ifcfg-$iface);
+    assert_script_run "rcnetwork restart";
+    assert_script_run "ip addr";
+    assert_script_run "ping -c 1 $gw || journalctl -b --no-pager > /dev/$serialdev";
+}
+
+sub can_upload_logs {
+    return (script_run('ping -c 1 ' . host_ip()) == 0);
```

Comment:
I am against oneliners especially when they used in single place. 
But yeah I know it is a matter of taste and we have different :)

Reply:
Yeah, actually I had an idea to do something more if we were not able to recover network and redirect file content to the serial, so there will be a second usage, I guess =)